File: CityLocal/CityLocal.API/Interfaces/IUserRepository.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using CityLocal.API.Models;

namespace CityLocal.API.Interfaces
{
    public interface IUserRepository
    {
        int Add(User user);
        int Update(User user);
        int Remove(int id);
        User Find(int id);
        IEnumerable<User> GetAll();
    }
}

File: API/CL.API/CL.API/Interfaces/IUserRepository.ts
import * as model from "../Models/userModel";

export interface IUserRepository {
    find(id: number,res:(res:model.UserModel)=>void): void;
    create(user:model.UserModel,res:(res:model.UserModel)=>void): void;
    update(user:model.UserModel,res:(res:model.UserModel)=>void): void;
    remove(id:number, res: (res: model.UserModel) => void): void;
}